src/InfoMCStructHelper.cc

```diff
@@ -125,20 +131,25 @@ namespace mu2e {
     fillGenInfo(bestprimarysp, geninfo);
   }
 
+  
   void InfoMCStructHelper::fillGenInfo(const art::Ptr<SimParticle>& gp, GenInfo& geninfo) {
+    if(gp.isNonnull()){
```

Comment:
When the genealogy is truncated at Parent, is the GP ptr null or invalid?

Reply:
So this `gp` should be renamed because it refers to the SimParticlePtr that is the argument of this function. To answer your question though, the truncated particle will have a valid GenParticle ptr